src/guest_memory.rs

```diff
@@ -859,31 +854,33 @@ impl<T: GuestMemory> Bytes<GuestAddress> for T {
 
     /// # Examples
     ///
-    /// * Write 128 bytes to /dev/null
+    /// * Write 128 bytes to /dev/null (uses the `backend-mmap` feature)
     ///
     /// ```
-    /// # #[cfg(feature = "backend-mmap")]
-    /// # use vm_memory::{Bytes, GuestAddress, mmap::GuestMemoryMmap};
+    /// ##[cfg(not(unix))]
+    /// # extern crate vmm_sys_util;
+    /// ##[cfg(feature = "backend-mmap")]
+    /// # {
+    /// # use vm_memory::{Bytes, GuestAddress, GuestMemoryMmap};
+    /// #
+    /// # let start_addr = GuestAddress(0x1000);
+    /// # let gm = GuestMemoryMmap::from_ranges(&vec![(start_addr, 1024)])
+    /// #    .expect("Could not create guest memory");
+    /// # let mut file = if cfg!(unix) {
     /// # use std::fs::OpenOptions;
-    /// # use std::path::Path;
-    ///
-    /// # #[cfg(all(unix, feature = "backend-mmap"))]
-    /// # fn test_write_null() {
-    ///     let start_addr = GuestAddress(0x1000);
-    ///     let gm =
-    ///         GuestMemoryMmap::from_ranges(&vec![(start_addr, 1024)])
-    ///         .expect("Could not create guest memory");
-    ///     let mut file = OpenOptions::new()
-    ///         .write(true)
-    ///         .open("/dev/null")
-    ///         .expect("Could not open /dev/null");
+    /// let mut file = OpenOptions::new()
+    ///     .write(true)
+    ///     .open("/dev/null")
+    ///     .expect("Could not open /dev/null");
+    /// #   file
+    /// # } else {
```

Comment:
The example doesn't look any different on windows, it just runs different. The way I look at this example now is just that since Windows doesn't support `/dev/null`, `TempFile` is a rough equivalent that enables some runtime test coverage of this example on Windows.

The thing I have against `TempFile` is that it's not a structure I would expect users to be familiar with or care about, since it's only really used internally in vmm tests. Using it in as a visible part of an example distracts from the actual purpose of the example. I personally got confused enough by this example at first glance to waste time looking up first where `TempFile` was defined, and second what it actually did.

I don't think either is necessary, since almost no examples I've seen show both the unix and windows cases, but I'd be more amenable to something like a call to a hidden function (e.g. `get_temp_file`).